task_processing/plugins/mesos/execution_framework.py

```diff
@@ -150,39 +148,34 @@ def offer_matches_pool(self, offer):
     def kill_task(self, task_id):
         self.driver.killTask(Dict(value=task_id))
 
-    def blacklist_slave(self, agent_id):
+    def blacklist_slave(self, agent_id, timeout):
         with self._lock:
```

Comment:
blacklist_slave is only called from within kill_tasks_stuck_in_staging which already acquires the lock